engines/mohawk/myst_state.cpp

```diff
@@ -251,12 +257,42 @@ bool MystGameState::saveMetadata(int slot) {
 	return true;
 }
 
+void MystGameState::isAutoSaveAllowed(void) {
+	// Open autosave slot and see if it an autosave
+	// Autosaving will be enabled if it is an autosave or if there is no save in that slot
+	Common::String filename = buildMetadataFilename(Mohawk::kAutoSaveSlot);
+	Common::InSaveFile *metadataFile = g_system->getSavefileManager()->openForLoading(filename);
+	if (!metadataFile) { // There is no save in the autosave slot, enable autosave
+		_isAutosave=true;
+		return;
+	}
+
+	Common::Serializer m(metadataFile, nullptr);
+
+	// Read the metadata file
+	Mohawk::MystSaveMetadata metadata;
+	if (!metadata.sync(m)) { // the save in the autosave slot is corrupted, enable autosave
+		_isAutosave=true;
+		delete metadataFile;
+		return;
+	}
+
+	if (metadata.autoSave == (uint8)1) {
```

Comment:
metadata.autoSave never is true. I expect this to be true when I have saved an autosave. Doesn't matter if I hardcode _metadata.autoSave = 1 (even for all slots) in MystGameState::updateMetadateForSaving. So maybe MystSaveMetadata::sync isn't loading/syncing the autoSave value because I am not saving saves with the new version (2)?

Reply:
It seems to be working for me.

Reply:
Does it work if you exit out of scummvm and restart it (with an autosave in slot 0)?

Reply:
I think so. But as I said it would be simpler and safer to just check if the save in slot 0 is an autosave just before trying to autosave than doing it on engine startup and keeping the value around.